spec/unit/puppet/provider/gpo/lgpo_spec.rb

```diff
@@ -60,6 +60,25 @@ def stub_create(scope, content, cse)
         provider.class.expects(:lgpo).once.with(*args).returns(nil)
         expect(File).to receive(:delete).once.with(out_polfile).and_return(nil)
     end
+    def hash_delete(scope, content, cse)
+        file = StringIO.new
+        allow(File).to receive(:open).once.with(out_file, 'w').and_yield(file)
+        allow(file).to receive(:write).with(content)
+
+        stub_lgpo_pol(scope, true)
+        allow(file).to receive(:open).at_least(:once).with(out_file, 'a').and_yield(file)
```

Comment:
This is `expect(File)`, not `expect(file)` since we're stubbing the `File` module, not the `file` object.